src/osyris/core/vector.py

```diff
@@ -252,3 +252,65 @@ def cross(self, other):
         z = self.x * other.y
         z -= self.y * other.x
         return self.__class__(x, y, z)
+
+    def _get_parent_pos(self):
+        group = self.parent
+        return group.get("position", group.parent["amr"]["position"])
+
+    def _get_colatitude(self, pos):
+        colatitude = np.arctan2(np.linalg.norm([pos.y.values, pos.x.values], axis=0),
```

Comment:
Why go via `numpy` here? Could you you make a 2d vector and call `norm` on that? `Vector(pos.x, pos.y).norm`.

However, I don't really understand why you use the formular with `arctan`, instead of `arccos(z / r)`? Would be much simpler?

Reply:
I'll try to see if your numpy wrapper works with atan2. This formula handles all degenerate solutions as well as the r=0 case for us. It's a cleaner syntax since it avoids countless if/else statements.

Reply:
But your expression also breaks down when z=0?

Reply:
No it does not. It only uses the x and y coordinates.

Reply:
Sorry, it does use the z coordinate, but atan2 handles that specific case for us as well.

Reply:
[See the following.](https://numpy.org/doc/stable/reference/generated/numpy.arctan2.html)

Reply:
I know how the `atan2` function works, but what do you do when `z=0`? There should only be one cell with `r = 0`, and if no theta angle can be defined there, I can live with that, because it doesn't make much physical sense anyway (since the radius is 0). However, there could be many cells with `z=0` in the midplane, and their theta should just be `pi/2`, not undefined.

Reply:
Oh, I had misunderstood you then. Yes, `arctan2(f(x,y), z)` does return `pi/2` when `z=0`.
See the following example:
```
x = np.random.rand(10)
y = np.zeros_like(x)

print(np.arctan2(x,y))

Out: [1.57079633 1.57079633 1.57079633 1.57079633 1.57079633 1.57079633
 1.57079633 1.57079633 1.57079633 1.57079633]

```

Reply:
> Why go via numpy here? Could you you make a 2d vector and call norm on that? Vector(pos.x, pos.y).norm

Just checked, yes I can and yes I will.

Reply:
Also, is there a reason why you'd rather we use your numpy wrapper on Vector rather than go through numpy directly?

Reply:
I apologize about the atan2, I simply misread the code, I thought you were dividing the cylindrical radius by z before sending to atan, but you are just passing it as the second argument to atan2 (which is exactly what atan2 was designed to do).

As for the use of numpy.linalg.norm, I think you can maybe just use the `.cyl_r` property of pos that you added. I don't think it uses the `_get_colatitude` function internally? Conceptually, that's exactly what you are tryin to extract, right?

Reply:
In the link that I sent you, there's this table:
![image](https://user-images.githubusercontent.com/60321972/177870763-db76a69e-d428-4ebb-9c6b-375de65dfc3c.png)
For the` x=y=0` case, `atan2` returns 0 (which means `theta` or `phi` = 0)

Reply:
👍 (the definition on Wikipedia just says it's undefined)